lightning/src/util/events.rs

```diff
@@ -974,13 +1005,14 @@ impl Writeable for Event {
 				error_data.write(writer)?;
 				write_tlv_fields!(writer, {
 					(0, payment_hash, required),
-					(1, network_update, option),
+					(1, None::<NetworkUpdate>, option), // network_update in LDK versions prior to 0.0.114
```

Comment:
Should we write the `NetworkUpdate` if we have a `Failure` with one?

Reply:
We could, I don't have a strong opinion. It's not horrible to break forwards compat here, old versions will just have `None` as their `network_update`. The downside is possibly double-(de)serializing a `ChannelUpdate` + its signature and might make the deser code a bit uglier